infoschema/cache.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infoschema
+
+import (
+	"sort"
+	"sync"
+)
+
+// InfoCache handles information schema, including getting and setting.
+type InfoCache struct {
+	// mu can not be RWMutex, because we access infoschema in the cache slice
+	mu sync.Mutex
+	// cache is sorted by SchemaVersion in descending order
+	cache []InfoSchema
+}
+
+// NewCache creates a new Handle.
+func NewCache(capcity int) *InfoCache {
+	return &InfoCache{cache: make([]InfoSchema, 0, capcity)}
+}
+
+// Get gets the newest information schema.
+func (h *InfoCache) Get() InfoSchema {
```

Comment:
InfoCache.Get() sounds like returning the cache. How about calling it GetLatest or something like that?

Reply:
Done